vvenancio297/library
# ENHANCEMENT: Change the whole flow of the reservation

# Description

I was not confortable with the quality of the app, so I decided to change some things:
- To avoid duplicated reservation codes, a code column that has an UUID was created
- In the front-end side, if the book is `returned` the admin should not be able to edit it
- The flash messages were all filled
- Some CSS fixes 
- I decided to remove the option for deleting a book, because it's better to just soft deleted it
- Reservation services were made to make the code better
- Quantity attribute was removed from the book, it's better to handle each book by itself

## How to test it
- Access the application as an admin, and create a book; log out
- Access the application as a member,  and make a reservation; you should be able to see the reservation code when the reservation succeed; log out
- Access the application as an admin and change the reservation status

---
__P.S.: Sorry for the amount of changes, I now it's not a best practice, but I really wanted to deliver a great quality code in the time I had, so I couldn't break it into smaller peaces of changes.__
